app/api/client.py

```diff
@@ -103,27 +102,20 @@ def _get_client(self, *, fetch_schema_from_transport: bool = False) -> Client:
     def logger(self) -> typedefs.Logger:
         return self._logger
 
-    def execute_gql(
-        self, document: DocumentNode, variable_values: dict[str, Any] | None = None
-    ):
+    def execute_gql(self, query: GraphQLRequest):
```

Comment:
Why did we have to change the signature here, and all the code calling this, rather than just create the request object inside?